docs/shared/recipes/plugins/project-graph-plugins.md

```diff
@@ -75,6 +75,30 @@ export const createNodes: CreateNodes = [
 ];
 ```
 
+{% callout type="warning" title="Dynamic target configurations can't be migrated" %}
+
+If you create targets for a project within a plugin's code, the Nx migration generators can not find that target configuration to update it. There are two ways to account for this:
+
+1. Only create dynamic targets using executors that you own. This way you can update the configuration in both places when needed.
+2. If you create a dynamic target for an executor you don't own, only define the `executor` property and instruct your users to define their options in the `targetDefaults` property of `nx.json`. The code would look something like this:
+
+```ts
+export const createNodes = ['**/jest.config.ts', (configFile, ctx) => {
```

Comment:
I wasn't sure whether we should show this code. It initially confused me because I wasn't sure what it is conveying until I realized it was to show the "only specifying executor".
That might be just me though, we can leave it :)